src/javascript/tests.rs

```diff
@@ -3,6 +3,34 @@ mod functions;
 mod numbers;
 mod strings;
 
+fn rocket_ship_module() -> crate::type_::Module {
+    let src = r#"
+pub const c = 299_792_458
+
+pub fn launch() {
+  Ok("launched")
+}
+pub fn fuel(amount: Int) {
+  Ok("fueled")
+}
+  "#;
```

Comment:
Would it be possible to instead create a macro that allows us to compile multiple modules? I would prefer this because then all of the data for the test can be viewable in one place rather than having to to know that there is some implicitly added module somewhere else in the codebase. I think this would make it much clearer what is happening in each test, and make it easier to create the required state in other tests.

This is what we currently do for the Erlang testing, though admittedly this code is not within the erl tests module

Reply:
I was getting a bit lost in rust macro land, so I stuck with this solution as it was working, certainly not very elegant.
Do you have some pointers for how you'd like it to work. some example erl code that does similar things.

Reply:
Maybe slap a TODO on this and I'll tackle this one after merge. No point wasting your time

Reply:
I'll add the TODO, I can always look myself later if I want to go deeper on rust macros

Reply:
Comment added